### DISCUSSION TITLE
Godot reverse audio (maybe not first one)

### DISCUSSION THREAD

**Discussion by foxp552** [godotengine/godot-proposals#10525] [Scripting]
There is a way to reverse audio with that ?
Audioplayer.seek(audioplayer.get_playback_positio() - 0.03)

**Comment by Meorge** [comment]
Are you looking/asking for a function like:
```gdscript
# inside of AudioStreamPlayer

func seek_by(amount_secs: float) -> void:
    seek(get_playback_position() + amount_secs)
```
Such that you could write code like
```gdscript
$AudioStreamPlayer.seek_by(3.0)  # moves 3 seconds ahead
$AudioStreamPlayer.seek_by(-2.5)  # moves 2.5 seconds back
```
? Or are you looking for something else?

**Reply by foxp552** [reply]
Kinda...
But better set input sec to 0.03 or around it for reverse playing
and it jus worked perfectly 🙂
Soo we can have some fun 😂

**Reply by Meorge** [reply]
What kinds of use cases would you have for shifting the playback position forward and backward by a certain amount? There's more or less only one use case that occurs to me, so as a result I think it might make more sense just to write your own helper method like
```gdscript
func seek_by(amount_secs: float) -> void:
    $AudioStreamPlayer.seek($AudioStreamPlayer.get_playback_position() + amount_secs)
```
to use, rather than add it to the `AudioStreamPlayer` class itself where it would be seldom used.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/3207.
